### PR TITLE
Handle game chats and write them into the log

### PR DESCRIPTION
**Pull request by windo**
Players sometimes say things in the game chat, which I would be interested in at least becoming aware of.

This ought to address https://github.com/online-go/gtp2ogs/issues/258

### CODE REVIEW

**Comment by windo** [comment]
I'm not entirely confident that it wouldn't be a better idea to still try to just `connectToGame` from main and make sure `Game` can see the game is finished and will disconnect?

Also, should this be optional?

**Comment by windo** [comment]
Oof, I've messed something up and it doesn't pick up chats during the game, will check later.

**Comment by windo** [comment]
Ought to be fixed. I'm running it on my bots now and they log when I chat with them during or after a game.

File: src/Game.ts [hunk 1]
```diff
@@ -827,6 +848,20 @@ export class Game extends EventEmitter<Events> {
...
 }
 
+export function handleChatLine(game_id: string, line: any, cutoff_timestamp: number) {
+    if (typeof line.body !== "string") {
+        return;
+    }
+    if (line.username === config.username) {
+        return;
+    }
+    if (line.date < cutoff_timestamp) {
```
Comment (raylu) [inline src/Game.ts:858]: dunno if it's worth noting that converting `new Date()` to a number gives you epoch time in UTC and `line.date` is also epoch time in UTC

File: src/Game.ts [hunk 2]
```diff
@@ -341,8 +344,26 @@ export class Game extends EventEmitter<Events> {
...
+                // game has already been dealt with.
+                handleChatLine(game_id, d.line, this.startup_timestamp);
+            };
+            socket.on(`game/${game_id}/chat`, on_chat);
+            this.on("disconnecting", () => {
+                socket.off(`game/${game_id}/chat`, on_chat);
+            });
+        }
+
+        trace.log("connecting");
```
Comment (raylu) [inline src/Game.ts:363]: probably remove this or mention what game we're connecting for if you keep it

File: src/Game.ts (hunk 1 above)
Comment (windo) [reply src/Game.ts:858]: If you felt like asking, I'm happy to add a note!

File: src/Game.ts (hunk 2 above)
Comment (windo) [reply src/Game.ts:363]: Whoops, yes, this was a leftover from debugging, will remove!

**Pull request merged**